x-pack/plugins/security_solution/common/api/detection_engine/prebuilt_rules/perform_rule_upgrade/perform_rule_upgrade_route.schema.yaml

```diff
@@ -0,0 +1,865 @@
+openapi: 3.0.0
+info:
+  title: Perform Rule Upgrade API endpoint
+  version: '2023-10-31'
+paths:
+  /api/detection_engine/rules/prebuilt/_perform_upgrade:
+    post:
+      x-labels: [ess]
+      x-codegen-enabled: true
+      operationId: PerformRuleUpgrade
+      summary: Perform rule upgrade
+      description: Upgrade prebuilt detection rules.
+      tags:
+        - Rules API
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              oneOf:
+                - $ref: '#/components/schemas/UpgradeAllRulesRequest'
+                - $ref: '#/components/schemas/UpgradeSpecificRulesRequest'
+      responses:
+        200:
+          description: Indicates a successful call.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/PerformRuleUpgradeResponseBody'
+
+components:
+  schemas:
+    PickVersionValues:
+      type: string
+      enum: [BASE, CURRENT, TARGET, MERGED]
+
+    RuleUpgradeSpecifier:
+      type: object
+      required:
+        - rule_id
+        - revision
+        - version
+      properties:
+        rule_id:
+          $ref: '../../model/rule_schema/common_attributes.schema.yaml#/components/schemas/RuleSignatureId'
+        revision:
+          type: number
+        version:
+          $ref: '../../model/rule_schema/common_attributes.schema.yaml#/components/schemas/RuleVersion'
+        pick_version:
+          $ref: '#/components/schemas/PickVersionValues'
+        fields:
+          type: object
+          description: |
+            Fields that can be customized during the upgrade workflow
+            as decided in: https://github.com/elastic/kibana/issues/186544
+            Fields listed here, which are not specified in the request body,
+            will default to a `pick_version` of `MERGED`.
```

Comment:
Descriptions become a part of our public documentation. We should avoid putting technical details or links to GitHub issues in the description.

Also, is this part still accurate? 

```
Fields listed here, which are not specified in the request body,
will default to a `pick_version` of `MERGED`.
```

Shouldn't omitted fields default to the high-level `pick_version`?

Reply:
> Shouldn't omitted fields default to the high-level pick_version?

Yes and no. Omitted fields will default to the higher-level `pick_version`, but only if one is defined (notice that `pick_version` is not `required` at any higher level, only on each field within `fields`). So I think the behaviour should be:

1. Use the `pick_version` defined within each rule field.
2. If the field is not defined, update that field to:
  2.1. `MERGED` if no higher-level `pick_version` is defined.
  2.2. Otherwise: the `pick_version` passed on the `RuleUpgradeSpecifier`, if defined.
  2.3. Otherwise: the `pick_version` passed on the `UpgradeSpecificRulesRequest`, if defined.
  2.4. Otherwise (no `pick_version` passed on individual fields or at any level): use `MERGED` as default.

Same for `UpgradeAllRulesRequest`: here use `MERGED` as default, unless `pick_version` is explicitly passed.

What do you think? The alternative to this would be to make the top-level `pick_version` required so that we can default to that instead of `MERGED`.